platform/engineering/design-docs/2021-05-18-forms-library.md

```diff
@@ -0,0 +1,852 @@
+The book _Good Services_ sets forth a couple of relevant service design
+principles:
+- Work in a way that is familiar
+- Require no prior knowledge to use
+
+The form configuration approach is not one that is familiar to front end
+engineers to create applications and requires _much_ prior knowledge to use
+effectively. When learning the basics of the library, its complexity is
+relatively minimal, but quickly grows as the form becomes more complex. This is
+largely due to the library bloat outlined above.
+
+##### And more!
+There are many other problems introduced by the existing forms library, but for
+now, I digress.
+
+### High Level Design
+Formulate will be a modular library consisting of helper functions and
+components to aid in the rapid development of forms on VA.gov. These functions
+and components can be conceptualized as **sub-modules** for:
+- [Building form pages](#form-page-builder)
+  - Wrappers for Formik components using the design system
+  - Re-exporting necessary Formik components
+- [Routing](#routing)
+  - Simple set-up for using Formik across multiple pages
+  - Components for rendering the standard form "chrome"
+  - Looping through a series of pages per item in an array
+  - Conditionally-rendered pages
+- [Save in progress](#save-in-progress)
+  - Manual save
+  - Auto save
+  - Data migrations
+
+**Each sub-module will be independent.** If a form doesn't need the multi-page
+routing, the team building it simply won't use the routing sub-module, and it
+won't be included in their application bundle. Likewise, if a form needs more
+complex routing, the team building it can opt to roll their own, but still reap
+the benefits of using the rest of the system.
+
+_Note: "sub-modules" here refer to independent parts of a larger system and have
+nothing to do with [Git
+Sub-Modules](https://git-scm.com/book/en/v2/Git-Tools-Sub-Modules)._
+
+Additionally, the library will provide a number of helper functions and
+components for:
+- Field-level schema validation
+- [Building a review page](#json-schema-validation)
+
+## Specifics
+
+### Detailed Design
+**Key technologies** we'll use:
+- React
+- TypeScript
+- [`api-extractor`](https://api-extractor.com/pages/overview/intro/)
+- [React router](https://reactrouter.com/)
+- [Formik](https://formik.org/)
+
+The following sections outline the various parts of the system.
+
+#### Form page builder
+Formulate will leverage Formik for much of the form page building. It will
+**re-export Formik components** where it makes sense and **provide light
+wrappers for Formik input fields** using design system components.
+
+In addition to simple wrappers, Formulate will use Formik to provide more
+complex fields such as file upload which may be imported and used in
+applications.
+
+##### Why not just use Formik directly?
+It should be possible to use Formik directly without any issues. The form page
+builder sub-module intends to make using the design system with Formik simpler.
```

Comment:
So there are two answers to your question.

**Philosophically**
It will not be the responsibility of Formulate nor the design system team to enforce the usage of only Formulate fields, if that's your question. We can highly recommend it, however, because of the benefits teams will enjoy in the form of:
- Simple usage (better DX)
  - I see now I need to add a bit in here about wrapping the label and error message as well; this is where the simpler usage comes in
- "Free" bug fixes, a11y improvements, and design updates (when applicable)

At the end of the day, if a team wants to totally deviate from the design system, they own their product and are responsible and accountable for it. :man_shrugging: We simply provide the tools to make design system adoption easy and desirable.

**Technically**
Also no. :sweat_smile: There's not really a way I can think of to stop folks from just using `<Field>`.